ftplugin/javascript/cfi.vim

```diff
@@ -156,13 +177,15 @@ function! s:finder.get_func_name() "{{{
         \ '\(function\)\s*\(\*\)\=\s*',
         \ '\1\2', '')
 
-  return (g:cfi_javascript_show_assignment ? l:assignment . ': ' : '')
-        \. l:variable_name
-        \. (!empty(l:function_type) ? l:function_type . ' ' : '')
-        \. l:function_name
-        \. l:function_arguments
-        \. l:function_body
-
+  " Use the setting dict to selectively show parts of the name
+  " Don't let the user break this, cycle through our dictionary for keys
+  let l:result = ''
+  for var in keys(s:show_defaults)
```

Comment:
Hmm, I think you cannot access directly to `l:`
because `l:assignment` and `l:function_type` has extra strings (`': '` and ` `).
You must construct `l:result` string by `:if` each by each.

And this is very small thing though, please tell me why you use `s:show_defaults`?
if you feel afraid accessing global variable because user might destroy it, you should `get(g:cfi_javascript_show, var, 0)` and so on.
Or not, just use `g:cfi_javascript_show`.